tests/rptest/tests/data_migrations_api_test.py

```diff
@@ -214,3 +218,207 @@ def test_creating_and_listing_migrations(self):
+        producer.start(clean=False)
+
+        wait_until(lambda: producer.produce_status.acked > 10,
+                   timeout_sec=120,
+                   backoff_sec=1)
+        return producer
+
+    def start_consumer(self, topic):
+        consumer = KgoVerifierConsumerGroupConsumer(self.test_context,
+                                                    self.redpanda,
+                                                    topic,
+                                                    self.msg_size,
+                                                    readers=3,
+                                                    group_name="test-group")
+
+        consumer.start(clean=False)
+        return consumer
+
+    def validate_topic_access(
+        self,
+        topic: str,
+        metadata_locked: bool,
+        read_blocked: bool,
+        produce_blocked: bool,
+        test_add_partitions: bool = True,
+    ):
+        rpk = RpkTool(self.redpanda)
+        assert topic in list(
+            rpk.list_topics()), f"validated topic {topic} must be present"
+        if test_add_partitions:
+            # check if partitions can be added to the topic
+            try:
+                self.logger.info(f"trying to add partition to {topic}")
+                rpk.add_topic_partitions(topic, 33)
+                assert not metadata_locked, "It shouldn't be possible to add partitions to the topic"
+            except Exception as e:
+                self.logger.info(f"adding partitions to {topic} failed")
+                assert metadata_locked, "It should be possible to add partitions to the topic"
+
+        # check if topic properties can be altered
+        try:
+            self.logger.info(f"trying update {topic} properties")
+
+            rpk.alter_topic_config(topic, TopicSpec.PROPERTY_FLUSH_MS, 2000)
+            rpk.delete_topic_config(topic, TopicSpec.PROPERTY_FLUSH_MS)
+            assert not metadata_locked, "Topic properties should not be updatable"
+        except Exception as e:
+            self.logger.info(f"updating {topic} properties failed - {e}")
+            assert metadata_locked, "Topic properties should be updatable"
+
+        # check if topic is readable
+        try:
+            rpk.consume(topic=topic, n=1, offset=0)
+            assert not read_blocked, "Topic should not be readable"
+        except Exception as e:
+            self.logger.info(f"consuming from {topic} failed - {e}")
+            assert read_blocked, "Topic should be readable"
+
+        # check if topic is writable only if it is expected to be blocked not to disturb the verifiers.
+        if produce_blocked:
+            try:
+                rpk.produce(topic=topic, key="test-key", msg='test-msg')
+                assert not produce_blocked, "Topic writes should be blocked"
+            except Exception as e:
+                self.logger.info(f"produce to {topic} failed - {e}")
+                assert produce_blocked, "Topic writes should be possible"
```

Comment:
This check cannot fail.

In general the try-except with two asserts only fails if an operation is blocked but expected to be allowed, but not the other way round. That's because AssertionError is an Exception, so the except block captures an assertion in the try block. This can be fixed either with narrowing down the exception class matcher, or something like this:
```
	try:
		attempt_op()
		blocked = False
	except Exception:
		# logging here if needed
		blocked = True
	assert blocked == expected, "some more logging"
```